sc-tools/sc-builder/tests/units/test_gwf_translator.cpp

```diff
@@ -4,33 +4,209 @@
+#include "sc_scs_tree.hpp"
+
+using GWFTranslatorTest = ScBuilderTest;
+
+#define BASE_TEST_PATH SC_BUILDER_KB "/tests-gwf-to-scs/"
+
+std::vector<char> ReadFileToBytes(std::string const & filePath)
+{
+  std::ifstream file(filePath, std::ios::binary | std::ios::ate);
+  std::streamsize size = file.tellg();
+  file.seekg(0, std::ios::beg);
+  std::vector<char> buffer(size);
+  file.read(buffer.data(), size);
+  return buffer;
+}
+
+bool CompareFiles(std::string const & filePath1, std::string const & filePath2)
+{
+  std::vector<char> file1Bytes = ReadFileToBytes(filePath1);
+  std::vector<char> file2Bytes = ReadFileToBytes(filePath2);
+  return file1Bytes == file2Bytes;
+}
+
+std::string ReadFileToString(std::string const & filePath)
+{
+  std::ifstream file(filePath);
+  std::stringstream buffer;
+  buffer << file.rdbuf();
+  return buffer.str();
+}
+
+bool CheckGWFToSCSTranslation(std::unique_ptr<ScAgentContext> & context, std::string const & fileName)
+{
+  GWFTranslator translator(*context);
+
+  std::string const & testGWFFilePath = BASE_TEST_PATH + fileName;
+  std::string const & correctSCsFilePath = testGWFFilePath + ".scs";
+
+  std::string const & testGWFText = translator.GetXMLFileContent(testGWFFilePath);
+  std::string const & testSCsText = translator.TranslateGWFToSCs(testGWFText, BASE_TEST_PATH);
+
+  std::string const & correctSCsText = ReadFileToString(correctSCsFilePath);
+
+  SCsTree testSCsTree;
+  testSCsTree.Parse(testSCsText);
+  SCsTree correctSCsTree;
+  correctSCsTree.Parse(correctSCsText);
+  return testSCsTree.Compare(correctSCsTree)->empty();
+}
+
+TEST_F(GWFTranslatorTest, InvalidPath)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const filePath = BASE_TEST_PATH "invalid_path.gwf";
+  EXPECT_THROW(translator.GetXMLFileContent(filePath), utils::ExceptionParseError);
+}
 
-TEST_F(ScBuilderTest, CorrectGWF)
+TEST_F(GWFTranslatorTest, EmptyFile)
 {
   GWFTranslator translator(*m_ctx);
 
-  Translator::Params translateParams;
-  translateParams.m_fileName = TEST_CORRECT_GWF;
-  translateParams.m_autoFormatInfo = true;
-  translateParams.m_outputStructure = ScAddr::Empty;
-  EXPECT_TRUE(translator.Translate(translateParams));
+  std::string const filePath = BASE_TEST_PATH "empty_file.gwf";
+  EXPECT_THROW(translator.GetXMLFileContent(filePath), utils::ExceptionParseError);
 }
 
-TEST_F(ScBuilderTest, IncorrectGWF)
+TEST_F(GWFTranslatorTest, EmptyStatic)
 {
   GWFTranslator translator(*m_ctx);
 
-  Translator::Params translateParams;
-  translateParams.m_fileName = TEST_INCORRECT_GWF;
-  translateParams.m_autoFormatInfo = true;
-  translateParams.m_outputStructure = ScAddr::Empty;
-  EXPECT_THROW(translator.Translate(translateParams), utils::ExceptionParseError);
+  std::string const & filePath = BASE_TEST_PATH "empty_static.gwf";
+  std::string const & gwfText = translator.GetXMLFileContent(filePath);
+  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, BASE_TEST_PATH), utils::ExceptionParseError);
```

Comment:
```suggestion
  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, filePath), utils::ExceptionParseError);
```
shouldn't filePath be passed here? check other tests as well